diff --git a/src/rdk_debug_priv.c b/src/rdk_debug_priv.c
index ae28e95..d40b3ca 100644
--- a/src/rdk_debug_priv.c
+++ b/src/rdk_debug_priv.c
@@ -325,6 +325,9 @@ void rdk_dbg_priv_init(void)
     (void) log4c_appender_type_set(&log4c_appender_type_to_console);
     (void) log4c_appender_type_set(&log4c_appender_type_to_syslog);
     (void) log4c_appender_type_set(&log4c_appender_type_to_journal);
+    (void) log4c_appender_type_set(&log4c_appender_type_rollingfile);
+
+    (void) log4c_rollingpolicy_type_set(&log4c_rollingpolicy_type_sizewin);
 
 
     if (log4c_init())
